h4 tag change in font weight for fixing issue (#3696)

soham283 opened the pull request:
<!-- General PR guidelines:

Most PRs should be opened against the main branch.

If the change should also be in the most recent release, add the
corresponding "cherrypick-0.X" label; for example, "cherrypick-0.12", to the
original PR. Best practice is to open a PR for the cherry-pick yourself after
your original PR has been merged into the main branch. Once the cherry-pick PR
has merged, remove the cherry-pick label from the original PR.

Use one of the new content templates:
  - [Documentation](./template-docs-page.md) -- Instructions and a template that
    you can use to help you add new documentation.
  - [Blog](./template-blog-entry.md) -- Instructions and a template that
    you can use to help you post to the Knative blog.

Learn more about contributing to the Knative Docs:
https://github.com/knative/docs
 -->



## Proposed Changes <!-- Describe the changes the PR makes. -->

- changed font weight for h4 tag similar value with other heading tag like h3.
- For fixing issue #3696
- 
![knative_h4](https://user-images.githubusercontent.com/65550720/121420410-0b040b80-c93b-11eb-84d6-7afe926715f9.png)

omerbensaadon commented:
CC @snneji @csantanapr

julz commented on docs/stylesheets/extra.css:
```diff
@@ -68,3 +68,9 @@
+
+/* Giving h4 tag distinctive style because it is non-differntial with bold text */
+.md-typeset h4 {
+  font-weight: 400;
```
`font-weight: 400` is the same as normal text (see https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#values), so Im not sure this is really an improvement :(.

soham283 replied:
The #3696 issue it was stated that bold text and h4 tags context must have distinct style so that it can be recognized easily
that's why I came up with this

julz replied:
yes, but this makes it so that the h4 tags are impossible to distinguish from regular text, which is even worse than being hard to distinguish from bold text?

soham283 replied:
maybe increasing font-size may work initially i thinked about that but thought if styling can make difference

soham283 commented:
Now updated the commit by changing font size, and removed font weight which were making is difficult to differentiate with the normal text.
@snneji @julz @omerbensaadon 
have a look on the new commit made

snneji commented:
Hi @soham283, thanks for working on this!

I think 1.0rem H4 font size looks a bit too large and heavy compared to H3 headings. I did a few tests in my browser. My personal preference is ~0.88rem. I was split between having the standard or a lighter font weight. My preferences are as follows:

- For the standard font weight (700), I prefer the H4 font size to be between 0.86 and 0.88 rem. Example page:
  <img width="526" alt="Comparative heading size 2 - weight 700" src="https://user-images.githubusercontent.com/33425661/121681411-b1bee800-cab2-11eb-833a-ba4526a309ed.png">
  <img width="529" alt="Headings within text - H4 weight 700" src="https://user-images.githubusercontent.com/33425661/121682390-f39c5e00-cab3-11eb-9b38-695314dc5a79.png">

- For a lighter font weight (400), I prefer the H4 font size to be a bit larger — between 0.88 and 0.9 rem. Example page:
  <img width="529" alt="Comparative heading size 3 - weight 400" src="https://user-images.githubusercontent.com/33425661/121681809-285be580-cab3-11eb-85f5-ad3b76768cb3.png">
  <img width="529" alt="Headings within text - H4 weight 400" src="https://user-images.githubusercontent.com/33425661/121682458-07e05b00-cab4-11eb-8f4b-6a55466fc73c.png">

I suppose this is a matter of personal taste so feel free to disagree @soham283, @julz, and @omerbensaadon!

csantanapr commented:
@snneji can you take another look when you have a chance?

snneji commented:
/lgtm

julz reviewed (approved):
/lgtm
/approve
